src/DynamoCore/Library/LibraryServices.cs

```diff
@@ -102,7 +160,7 @@ public string NicknameFromFunctionSignatureHint(string functionSignature)
 
             if (priorNameHints.ContainsKey(functionSignature))
             {
-                var mappedSignature = priorNameHints[functionSignature];
+                var mappedSignature = priorNameHints[functionSignature].UpgradeName;
```

Comment:
Are we guaranteed that this key exists? If not, trying to get the `UpgradeName` will blow up.